### PR TITLE
GRA-1030: update host name on join

### PR DESCRIPTION
**Pull request by abhishek9686**
refer PR https://github.com/gravitl/netclient/pull/156 for testing steps

### CODE REVIEW

**Review by theguy951357 — APPROVED** [review]
built server and nodes from netmaker and netclient branches.
joined network setting --name
netclient.yaml has new name in there
host on ui has new name
pass

File: controllers/node.go [hunk 1]
```diff
@@ -592,8 +592,15 @@ func createNode(w http.ResponseWriter, r *http.Request) {
 				logic.ReturnErrorResponse(w, r, logic.FormatError(err, "internal"))
 				return
 			}
-			logic.UpdateHost(&data.Host, host) // update the in memory struct values
-
+			logic.UpdateHostFromClient(&data.Host, host) // update the in memory struct values
+			logger.Log(0, fmt.Sprintf("HOST ----------> %+v\n", data.Host))
+			err = logic.UpsertHost(host)
```
Comment (mattkasun) [inline controllers/node.go:597]: this is going to overwrite the encoded password and cause issues going forward.

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: controllers/node.go (hunk 1 above)
Comment (abhishek9686) [reply controllers/node.go:597]: no it won't

Comment (0xdcarns) [reply controllers/node.go:596]: he is maintaining the current host in db

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**